### PR TITLE
ftr: error page for unauthorized users or page not found

### PR DESCRIPTION
**Pull request by ds-ext-dskrzypczak**
TRACEFOSS-859

### CODE REVIEW

File: src/app/modules/page/error-page/presentation/error-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import { RoleService } from '@core/user/role.service';
+import { LayoutFacade } from '@shared/abstraction/layout-facade';
+
+@Component({
+  selector: 'app-error',
+  templateUrl: './error-page.component.html',
+  styleUrls: ['./error-page.component.scss'],
+})
+export class ErrorPageComponent {
+  public title: string = 'errorPage.title';
```
Comment (evia-fbedarf) [inline src/app/modules/page/error-page/presentation/error-page.component.ts:34]: Put these translations keys into the html, and change the logic that it is done in the HTML.
Text-specific stuff should be handled in the HTML no in the code.

File: src/app/modules/page/error-page/presentation/error-page.component.html [hunk 2]
```diff
@@ -19,12 +19,23 @@
 SPDX-License-Identifier: Apache-2.0
 -->
 
-<div class="page-not-found-container">
-  <section class="page-not-found-content">
-    <h1 class="text-9xl mb-4">404</h1>
-    <h3 class="mb-8">{{ 'errorPage.pageNotFound.message' | i18n }}</h3>
-    <a href="#">
-      <app-button [variant]="'raised'" [color]="'primary'">{{ 'actions.back' | i18n }}</app-button>
+<div class="error-page-container">
```
Comment (evia-fbedarf) [inline src/app/modules/page/error-page/presentation/error-page.component.html:22]: Is it possible to create a reusable component out of that? So we can reuse them later in other pages, if for example data fails to load for a table or something.

File: src/app/modules/page/error-page/presentation/error-page.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+          } as unknown as ActivatedRoute,
+        },
+      ],
+      roles,
+    });
+
+  it('should render generic error page - with user role', async () => {
+    const { fixture } = await renderErrorPageComponent({ errorPage: {}, roles: ['user'] });
+
+    await fixture.whenStable();
```
Comment (evia-fbedarf) [inline src/app/modules/page/error-page/presentation/error-page.component.spec.ts:51]: What does this do?

**Review by evia-fbedarf — CHANGES_REQUESTED** [review]

File: src/app/modules/page/error-page/presentation/error-page.component.spec.ts (hunk 3 above)
Comment (ds-ext-dskrzypczak) [reply src/app/modules/page/error-page/presentation/error-page.component.spec.ts:51]: https://angular.io/api/core/testing/ComponentFixture#whenStable

I've found out that it helps a lot if you need to wait for some async activity.

File: src/app/modules/page/error-page/presentation/error-page.component.html (hunk 2 above)
Comment (ds-ext-dskrzypczak) [reply src/app/modules/page/error-page/presentation/error-page.component.html:22]: Yeah, I remember our discussion to be able to do it in the future. I have been thinking about that, and I think the better approach is to implement it in separate task for given example and purpose. It shouln't take much time to adapt the code.

**Review by evia-fbedarf — APPROVED** [review]

**Pull request merged**